src/state/opamFileTools.mli

```diff
@@ -117,3 +117,10 @@ val dep_formula_to_string: formula -> string
 (** Sort opam fields: author, tags, depexts, depends, depopts, conflicts,
     pin_depends, extra_files, extra_sources *)
 val sort_opam: OpamFile.OPAM.t -> OpamFile.OPAM.t
+
+(** Extract depexts from a single opam file *)
+val extract_depexts : env:OpamFilter.env -> OpamFile.OPAM.t -> OpamSysPkg.Set.t
+
```

Comment:
Opinionated: there is no need for the 2 functions, `extract_depexts` can take a map and if there is a single element a singleton can be created.